### PR TITLE
🪟 🐛   Update connection name - fix minor issues

### LINKED ISSUE #13416: Renaming connection name: couple of issues
## Environment
- **Airbyte version**:  0.39.7-alpha
- **OS Version / Instance**: macOS, Chrome 
- **Deployment**: OSS (Docker)

## Current Behavior
- When a user tries to rename the connection name, the text is aligned to the left in edit mode.
inpup_edit_mode_align_left

- User is unable to remove the whole name, the last character is left
last_character_connection_name

- User is able to create a connection with an empty name
empty_connection_name

## Expected Behavior
- IMO text in edit mode should be aligned to the center also since it's more natural.
- User should be able to remove the whole name
- If a user tries to save a connection name with an empty name - set the name which was before he started editing the connection name (TBD: @timroes )

### PR DESCRIPTION
**Pull request by dizel852**
## What
Closes [#13416](https://github.com/airbytehq/airbyte/issues/13416)

## How
- Fix the case when a user wasn't able to remove the last char in the connection name
- Remove the whole connection name by "selecting all" keyboard shortcut
- Do not allow the user to save an empty name - revert to the name set before the edit action.

CPT2206071129-1179x336

## Recommended reading order
1. `Input.tsx` - there was an issue in the base Input component. With extra props drilling which caused firing 'onChange' twice
2. `ConnectionName.tsx`
3. the rest

### CODE REVIEW

File: airbyte-webapp/src/components/base/Input/Input.tsx [hunk 1]
```diff
@@ -102,7 +102,7 @@ const Input: React.FC<InputProps> = (props) => {
   }, [inputRef, defaultFocus]);
 
   return (
-    <InputContainer {...props} className={focused ? "input-container--focused" : undefined}>
+    <InputContainer className={focused ? "input-container--focused" : undefined}>
       <InputComponent
         {...props}
```
Comment (dizel852) [inline airbyte-webapp/src/components/base/Input/Input.tsx:107]: @edmundito need your advice. Here we pass the props to `InputContainer` and `InputComponent`. Which causes firing `onChange` twice. Do we really need to rethrow props to `InputContainer`?

**Review by teallarson — COMMENTED** [review]
Nice fixes.  I did notice that we are able to edit the Connection name when the Connection is in the middle of a Sync.  When you do so, unfocusing the field doesn't reflect the "new" name (you have to refresh the page to see it).  I wonder if we should be able to edit the name during a sync or not.  If so, we should probably tidy that up!

**Comment by dizel852** [comment]
> if we should be able to edit the name during a sync or not.

@tealjulia Good question! Need to polish this feature as much as we can 🔥 
@timroes @edmundito @andyjih  any thoughts?

**Comment by andyjih** [comment]
I think editing the name of a connection is unrelated to sync jobs that might be in-progress, so I don't think of that as an issue.

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ConnectionName.tsx [hunk 2]
```diff
@@ -130,9 +42,16 @@ const ConnectionName: React.FC<Props> = ({ connection }) => {
...
   const updateConnectionAsync = async () => {
-    // Update only when the name is changed
-    if (connection.name !== connectionName) {
+    if (connectionName === undefined || connection.name === connectionName.trim() || !connectionName.trim()) {
+      setConnectionName(connectionNameBackup);
+      setEditingState(false);
+      return;
+    }
+
+    const connectionNameTrimmed = connectionName.trim();
```
Comment (edmundito) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ConnectionName.tsx:51]: You could use this var as the check:
```suggestion
    const connectionNameTrimmed = connectionName?.trim();
    if (!connectionNameTrimmed || connection.name === connectionNameTrimmed) {
      setConnectionName(connectionNameBackup);
      setEditingState(false);
      return;
    }
```

File: airbyte-webapp/src/utils/withKeystrokeHandler.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import * as React from "react";
+
+interface InProps {
+  readonly onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
```
Comment (edmundito) [inline airbyte-webapp/src/utils/withKeystrokeHandler.tsx:4]:
```suggestion
  readonly onKeyDown?: React.KeyboardEventHandler<HTMLInputElement>;
```

File: airbyte-webapp/src/scss/_variables.scss [hunk 4]
```diff
@@ -3,6 +3,9 @@ $transition: 0.3s;
 $border-thin: 1px;
 $border-thick: 2px;
 
+$border-radius-sm: 8px;
+$border-radius-m: 10px;
```
[the remark this answers is not in the document]
Comment (dizel852) [reply airbyte-webapp/src/scss/_variables.scss:7]: Ok, thanks for letting me know! Will merge with latest master

File: airbyte-webapp/.eslintrc [hunk 5]
```diff
@@ -4,9 +4,10 @@
     "plugin:@typescript-eslint/recommended",
     "plugin:jest/recommended",
     "prettier",
-    "plugin:prettier/recommended"
+    "plugin:prettier/recommended",
+    "plugin:css-modules/recommended"
   ],
-  "plugins": ["react", "@typescript-eslint", "prettier", "unused-imports"],
+  "plugins": ["react", "@typescript-eslint", "prettier", "unused-imports", "css-modules"],
```
[the remark this answers is not in the document]
Comment (dizel852) [reply airbyte-webapp/.eslintrc:10]: Agree, will move these changes to another PR

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ConnectionName.tsx (hunk 2 above)
Comment (dizel852) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ConnectionName.tsx:51]: Yeah, you are right! I will update this part 👍

File: airbyte-webapp/src/utils/withKeystrokeHandler.tsx (hunk 3 above)
Comment (dizel852) [reply airbyte-webapp/src/utils/withKeystrokeHandler.tsx:4]: This one also 👍

**Review by edmundito — APPROVED** [review]

**Pull request merged**